content-api/content-actors/src/main/scala/org/sunbird/content/util/RetireManager.scala

```diff
@@ -0,0 +1,120 @@
+                    val childIds = getChildrenIdentifiers(hierarchyMap)
+                    //TODO : Implement Delete Units from Elastic Search
+                    if (CollectionUtils.isNotEmpty(childIds)) {
+                        val topicName = Platform.getString("kafka.topics.graph.event", "sunbirddev.learning.graph.events")
+                        childIds.foreach(id => kfClient.send(ScalaJsonUtils.serialize(getLearningGraphEvent(request, id)), topicName))
+                        RedisCache.delete(childIds.map(id => "hierarchy_" + id): _*)
+                    }
+                    hierarchyMap.put("status", "Retired")
+                    hierarchyMap.put(HierarchyConstants.LAST_UPDATED_ON, DateUtils.formatCurrentDate)
+                    hierarchyMap.put(HierarchyConstants.LAST_STATUS_CHANGED_ON, DateUtils.formatCurrentDate)
+                    req.put("hierarchy", ScalaJsonUtils.serialize(hierarchyMap))
+                    ExternalPropsManager.saveProps(req)
+                }
+                ResponseHandler.OK()
+            }) recover { case e: ResourceNotFoundException =>
+                TelemetryManager.log("No hierarchy is present in cassandra for identifier:" + node.getIdentifier)
+                throw new ServerException("ERR_CONTENT_RETIRE", "Unable to fetch Hierarchy for Root Node: [" + node.getIdentifier + "]")
+            }
+        } else Future(ResponseHandler.OK())
+    }
+
+    private def getChildrenIdentifiers(hierarchyMap: util.HashMap[String, AnyRef]): util.List[String] = {
+        val childIds: ListBuffer[String] = ListBuffer[String]()
+        addChildIds(hierarchyMap.getOrElse(HierarchyConstants.CHILDREN, new util.ArrayList[util.HashMap[String, AnyRef]]()).asInstanceOf[util.ArrayList[util.HashMap[String, AnyRef]]], childIds)
+        bufferAsJavaList(childIds)
+    }
+
+    private def addChildIds(childrenMaps: util.ArrayList[util.HashMap[String, AnyRef]], childrenIds: ListBuffer[String]): Unit = {
+        if (CollectionUtils.isNotEmpty(childrenMaps)) {
+            childrenMaps.filter(child => StringUtils.equalsIgnoreCase(HierarchyConstants.PARENT, child.get(HierarchyConstants.VISIBILITY).asInstanceOf[String])).foreach(child => {
+                childrenIds += child.get(HierarchyConstants.IDENTIFIER).asInstanceOf[String]
+                addChildIds(child.get(HierarchyConstants.CHILDREN).asInstanceOf[util.ArrayList[util.HashMap[String, AnyRef]]], childrenIds)
+            })
+        }
+    }
+
+    private def getLearningGraphEvent(request: Request, id: String): Map[String, Any] = Map("ets" -> System.currentTimeMillis(), "channel" -> "in.ekstep", "mid" -> UUID.randomUUID.toString, "nodeType" -> "DATA_NODE", "userId" -> "Ekstep", "createdOn" -> DateUtils.format(new Date()), "objectType" -> "Content", "nodeUniqueId" -> id, "operationType" -> "DELETE", "graphId" -> request.getContext.get("graph_id"))
```

Comment:
Fetch channel from configuration
`"channel" -> "in.ekstep"`
from
default.channel